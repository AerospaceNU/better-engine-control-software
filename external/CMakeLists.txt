add_subdirectory(expected)
add_subdirectory(libserial)
