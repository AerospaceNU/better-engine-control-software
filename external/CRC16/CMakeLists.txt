add_library(CRC16 STATIC
    CRC16.cpp)
target_include_directories(CRC16 INTERFACE .)
