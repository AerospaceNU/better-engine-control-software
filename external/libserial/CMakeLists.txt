message(STATUS "Fetching LibSerial... (this will take a while)")


set(LIBSERIAL_ENABLE_TESTING OFF CACHE BOOL "Enables building unit tests" FORCE)
set(LIBSERIAL_BUILD_EXAMPLES OFF CACHE BOOL "Enables building example programs" FORCE)
set(LIBSERIAL_PYTHON_ENABLE OFF CACHE BOOL "Enables building the library with Python SIP bindings" FORCE)
set(LIBSERIAL_BUILD_DOCS OFF CACHE BOOL "Build the Doxygen docs" FORCE)
set(INSTALL_SHARED OFF CACHE BOOL "Install shared object library." FORCE)


#message(STATUS "LIBSERIAL_ENABLE_TESTING VAL: ${LIBSERIAL_ENABLE_TESTING}")
#option(LIBSERIAL_ENABLE_TESTING "Enables building unit tests" OFF)
#option(LIBSERIAL_BUILD_EXAMPLES "Enables building example programs" OFF)
#option(LIBSERIAL_PYTHON_ENABLE "Enables building the library with Python SIP bindings" OFF)
#option(LIBSERIAL_BUILD_DOCS "Build the Doxygen docs" OFF)

# make available
if (IS_CROSS)
    FetchContent_MakeAvailable(libserial)
endif()
