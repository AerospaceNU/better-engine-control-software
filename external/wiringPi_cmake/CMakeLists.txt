# this is actually a patch of WiringPi. WiringPi has been abandoned by the original creator, but there's still
# a GitHub mirror of the source. this guy patched the build system to be CMake instead of Makefiles, so we're using
# his branch

# it's worth mentioning that wiringPi is a LGPL library
# if you don't know what GPL is, it's a famous open source license for projects. the gimmick is copyleft,
# if your project uses a GPL library, you have to make your own project GPL/open source as well
# (for the record, I'm fine with this. the idea's neat as a way of making sure companies don't just
# leech off your work. the main reason I don't wanna do it is that I think all of Redshift's GitLab projects
# are private, and I don't really feel like finding the damn organization/repo setting to switch it public)

# LGPL is a bit different though. if you link to a shared library which is LGPL, you are still clear,
# and wiringPi builds as a shared library

# yeah, wiringPi compiles with these warnings triggering, which are enabled by default
# which is worrying, but fuck fixing that
add_compile_options("$<$<CXX_COMPILER_ID:GNU>:-Wno-int-to-pointer-cast;-Wno-pointer-to-int-cast>")
add_subdirectory(WiringPi)
