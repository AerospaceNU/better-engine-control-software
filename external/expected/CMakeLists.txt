# the expected cmakelist also give additional build targets that we switch off here
set(BUILD_TESTING OFF CACHE BOOL "Enables building unit tests" FORCE)
set(EXPECTED_BUILD_PACKAGE OFF CACHE BOOL "Enables building the package for various Linux distros" FORCE)

add_subdirectory(expected)
